editor.planx.uk/src/pages/FlowEditor/lib/store/navigation.ts

```diff
@@ -1,16 +1,106 @@
+import { TYPES } from "@planx/components/types";
+import { hasFeatureFlag } from "lib/featureFlags";
+import { findLast } from "lodash";
+import { Store } from "pages/FlowEditor/lib/store";
 import type { StateCreator } from "zustand";
 
+import { SharedStore } from "./shared";
+
+interface SectionNode extends Store.node {
+  data: {
+    title: string;
+  };
+}
+
 export interface NavigationStore {
-  navTest: string;
-  setNavTest: (navTest: string) => void;
+  currentSectionIndex: number;
+  sectionCount: number;
+  currentSectionTitle?: string;
+  hasSections: boolean;
+  sectionNodes: Record<string, SectionNode>;
+  initNavigationStore: () => void;
+  updateSectionData: () => void;
+  filterFlowByType: (type: TYPES) => Store.flow;
 }
 
 export const navigationStore: StateCreator<
-  NavigationStore,
+  NavigationStore & SharedStore,
   [],
   [],
   NavigationStore
-> = (set) => ({
-  navTest: "default value",
-  setNavTest: (navTest: string) => set({ navTest }),
+> = (set, get) => ({
+  currentSectionIndex: 1,
+
+  sectionCount: 0,
+
+  currentSectionTitle: undefined,
+
+  hasSections: false,
+
+  sectionNodes: {},
+
+  /**
+   * Set up initial values to populate store
+   * Called by setFlow() as we require a flow from the DB before proceeding
+   */
+  initNavigationStore: () => {
+    const sectionNodes = get().filterFlowByType(TYPES.Section) as Record<
+      string,
+      SectionNode
+    >;
+    const sectionCount = Object.keys(sectionNodes).length;
+    const hasSections = Boolean(
+      sectionCount && hasFeatureFlag("NAVIGATION_UI")
+    );
+    const currentSectionTitle = Object.values(sectionNodes)[0]?.data.title;
+
+    set({
+      sectionNodes,
+      sectionCount,
+      hasSections,
+      currentSectionTitle,
+    });
+  },
+
+  /**
+   * Update title and index on record()
+   * Triggered when going backewards, forwards, or changing answer
+   */
+  updateSectionData: () => {
+    const { breadcrumbs, sectionNodes, hasSections } = get();
+    // Sections not being used, do not proceed
+    if (!hasSections) return;
+
+    const breadcrumbIds = Object.keys(breadcrumbs);
+    const sectionIds = Object.keys(sectionNodes);
+    const mostRecentSectionId = findLast(
+      breadcrumbIds,
+      (breadcrumbId: string) => sectionIds.includes(breadcrumbId)
+    );
+
+    // This should not happen - if we have sections, we should have a most recent section
+    // Failing noisily for now will help identify issues whilst this is being developed & tested
+    // TODO: Fail more gracefully once feature is complete
+    if (!mostRecentSectionId) throw Error("Error finding mostRecentSectionId");
```

Comment:
This condition gets hit if a Section is not the first component in a flow, which I believe is one of the conditions we'll be validating against.

Reply:
This is a good shout and wasn't on our initial rules list, but is totally possible. It's a bit tricky because I'd naturally want to check for this on publish, but we actually need it valiated as soon as a section is added to the graph right? And I guess we'd also have to check that when a section is moved or deleted, if there are other sections, one of them is still satisfying the start position?

Reply:
So without this error, if we just say `if (!mostRecentSectionId) return` things work perfectly fine - whichever your first section is will get displayed first and will stay there until you pass into another section. This means we can chose not to enforce this if we don't want to, but it's just a bit of weird behaviour where the view the editor sees doesn't exactly match the behaviour which I'd rather avoid.

Ideally we'd catch this in the Editor (before publishing). For a plain `add()` I think it's not to bad....
- When adding a section, check if there's a section already at `_root.edges[0]`
  - If so, proceed
  - If not, but this component is being placed there, proceed
  - Otherwise, error
  
I think it's a little more complex on move or delete though like you said, as we'd need to run these checks when other node types and modified.

For now, can we get away with just checking on publish "if there's a section, is there a section at `_root.edges[0]`"?

Reply:
Sounds perfect 👍

Reply:
Validating we have a node in first root edge position on publish addressed here https://github.com/theopensystemslab/planx-new/pull/1481/commits/caaab6a37e165094ad320649605a3e2078f2ea07